xml_converter/generators/code_generator.py

```diff
@@ -287,27 +274,46 @@ def write_webdocs(self, output_directory: str) -> None:
         file_loader = FileSystemLoader('web_templates')
         env = Environment(loader=file_loader)
         template = env.get_template("documentation.html")
+        categories: Dict[str,List[str]] = {}
 
-        navigation_links = [ (self.data[x].metadata['attribute_category'], x) for x in sorted(self.data.keys()) ]
+        for filename in self.data.keys():
+            category = os.path.basename(os.path.dirname(filename))
+            if category not in categories:
+                categories[category] = []
+            categories[category].append(filename)       
+
+        navigation_links = [ (capitalize(category), category) for category in sorted(categories.keys()) ]
+        
+        
 
         complete_field_row_list = []
+        pages: Dict[str,List[str]] = {}
+
+        for page in sorted(categories):
+            content: Dict[str,List[str]] = {}
+            metadata: Dict[str,List[str]] = {}
+            # Resets the content and metadata to empty for each loop
 
-        for page in sorted(self.data.keys()):
-            content = self.data[page].content
-            metadata = self.data[page].metadata
+            for subpage in categories[page]:
+                
+                content[subpage] = self.data[subpage].content
```

Comment:
You can combine the markdown step into this line
```suggestion
                content[subpage] = markdown.markdown(self.data[subpage].content)
```